File: examples/csharp/Queue1.cs
using System;

class Queue1
{
    static void Main()
    {
        int[] queue = new int[10];
        int front = 0, rear = -1;
        rear = Enqueue(queue, rear, 5);
        rear = Enqueue(queue, rear, 10);
        rear = Enqueue(queue, rear, 15);
        Console.WriteLine(Dequeue(queue, ref front, rear));
        Console.WriteLine(Dequeue(queue, ref front, rear));
    }

    static int Enqueue(int[] q, int rear, int val)
    {
        q[++rear] = val;
        return rear;
    }

    static int Dequeue(int[] q, ref int front, int rear)
    {
        return front <= rear ? q[front++] : -1;
    }
}

File: examples/java/Queue1.java
public class Queue1 {
    static int[] arr = new int[100];
    static int front = 0, rear = -1, size = 0;

    static void enqueue(int x) {
        if (size < arr.length) {
            rear = (rear + 1) % arr.length;
            arr[rear] = x;
            size++;
        }
    }

    static int dequeue() {
        if (size == 0) return -1;
        int x = arr[front];
        front = (front + 1) % arr.length;
        size--;
        return x;
    }

    public static void main(String[] args) {
        enqueue(10);
        enqueue(20);
        enqueue(30);
        System.out.println(dequeue());
        System.out.println(dequeue());
    }
}
